### PR TITLE
feat(web) - compactify module list and more!

### PR DESCRIPTION
**Pull request by wendybujalski**
Screenshot 2024-05-01 at 6 19 43 PM

Also in this PR -
- Made components for a few redundant sections of code
- Replaced the Collapsible component with TreeNode in all places and removed it from the vue-lib
- Cleanup and comments in TreeNode to make it easier to use
- New styling for PillCounter and integration with TreeNode

### CODE REVIEW

File: app/web/src/components/NoSelectionDetailsPanel.vue [hunk 1]
```diff
@@ -57,13 +58,43 @@ import {
...
+import { useComponentsStore } from "@/store/components.store";
 import SidebarSubpanelTitle from "./SidebarSubpanelTitle.vue";
 import ChangesPanel from "./ChangesPanel.vue";
 import SecretsPanel from "./SecretsPanel.vue";
 
+const componentsStore = useComponentsStore();
 const changeSetStore = useChangeSetsStore();
 const actionsStore = useActionsStore();
+
+const diffs = computed(() => {
```
Comment (stack72) [inline app/web/src/components/NoSelectionDetailsPanel.vue:74]: We should remove diffs from this calculation - they are no longer on the panel. I removed them in #3675

Comment (wendybujalski) [reply app/web/src/components/NoSelectionDetailsPanel.vue:74]: ah okay! will do

Comment (wendybujalski) [reply app/web/src/components/NoSelectionDetailsPanel.vue]: done!

**Comment by wendybujalski** [comment]
bors merge

**Comment by wendybujalski** [comment]
bors retry

**Pull request merged**